R/add_ess_header.R

```diff
@@ -0,0 +1,89 @@
+#' Add ESS Table Header
+#'
+#' This function replaces the counts in the default header of
+#' `gtsummary::tbl_svysummary()` tables to counts representing the
+#' Effective Sample Size (ESS). See [`ess()`] for details.
+#'
+#' @param x (`tbl_svysummary`)\cr
+#'   Object of class `'tbl_svysummary'` typically created with `gtsummary::tbl_svysummary()`.
+#' @param header (`string`)\cr
+#'   String specifying updated header.
+#'   Review `gtsummary::modify_header()` for details on use.
+#'
+#' @returns a 'gtsummary' table
+#' @export
+#' @importFrom rlang .env .data
+#'
+#' @examplesIf rlang::is_installed(c("survey", "gtsummary", "cards", "cardx", "dplyr"))
+#' svy <- survey::svydesign(~1, data = nhefs_weights, weights = ~ w_ate)
+#'
+#' gtsummary::tbl_svysummary(svy, include = c(age, sex, smokeyrs)) |>
+#'   add_ess_header()
+#'
+#' gtsummary::tbl_svysummary(svy, by = qsmk, include = c(age, sex, smokeyrs)) |>
+#'   add_ess_header(header = "**{level}**  \nN = {n_unweighted}; \nESS = {round(n, 1)}")
+add_ess_header <- function(x, header = "**{level}**  \nESS = {round(n, 1)}") {
```

Comment:
If you'd like the default to round the ESS to one decimal place, I suggest using `format()` because if a number rounds to an integer, it won't show the `0` in the decimal place, e.g. `40` instead of `40.0`. You could also use `gtsummary::style_numbe(n, digits = 1)`.

Reply:
Good point. Is the base equivalent to the `style_number()` example roughly `format(round(n, 1), nsmall = 1)`? Seems like at that point I might as well use `style_number()`

Reply:
Oh I guess I don't need `round()`: `format(n, digits = 1, nsmall = 1)`